Userland/Applications/PixelPaint/Tools/BrushTool.cpp

```diff
@@ -185,9 +185,34 @@ GUI::Widget* BrushTool::get_properties_widget()
 
 NonnullRefPtr<Gfx::Bitmap> BrushTool::build_cursor()
 {
-    m_scale_last_created_cursor = m_editor ? m_editor->scale() : 1;
-    auto scaled_size = size() * m_scale_last_created_cursor;
+    float new_scale = m_editor ? m_editor->scale() : 1;
+    auto scaled_size = size() * new_scale;
     auto containing_box_size = 2 * scaled_size;
+    bool draw_ellipse = true;
+
+    // If we have an ImageEditor scaled_size should not excede diagonal length of the ImageEditor
+    if (m_editor) {
+        // FIXME: The ImageEditor diagonal size could be saved on the ImageEditor and exposed to optimize.
+        //        It would be nice if that was recalculated when the ImageEditor was resized
+        if (m_scale_last_created_cursor != new_scale) {
+            if (m_editor_previous_size[0] != m_editor->width() || m_editor_previous_size[1] != m_editor->height()) {
+                m_editor_previous_size[0] = m_editor->width();
+                m_editor_previous_size[1] = m_editor->height();
```

Comment:
If `m_editor_previous_size` was just a normal old `IntSize` all of this would just be:
```
if (m_editor_previous_size != m_editor.size()) {
     m_editor_previous_size = m_editor.size();
     ...
}